### PR TITLE
Fixes #767: Enhanced Error Handling and Unique Folder Output for DAO Creation

### PR DESCRIPTION
**Pull request by ashutoshpw**
## Summary

This PR introduces improvements to the DAO creation flow, specifically focusing on error handling and the management of output data. Each DAO creation request now generates its own unique folder, ensuring that outputs are isolated and traceable. This is particularly beneficial for maintaining robustness in the API's operation.

### Unique Output Folder Generation
Each DAO creation request now saves the output to a unique folder. This change enhances the robustness of the API by preventing conflicts and overlaps between simultaneous operations.

### Improved Error Handling
- The API's error handling has been refined. When the payload is incorrect or incomplete, the API will now return a `400 Bad Request` error, indicating that the client's request cannot be processed due to client error.
- In cases where a CLI error occurs, the API responds with a `500 Internal Server Error`. The response includes a context ID, providing a reference that can be used to further investigate and debug the issue. Ideally, these cases should be captured over a Sentry Instance.

## Code Refinements
- Updates to `.nvmrc` to specify the node version, ensuring consistent runtime environments.
- Modifications to the `Dockerfile` and `docker-compose.yml` to optimize the build process and development workflow. These changes include moving the `COPY` operation for better layer caching in Docker and utilizing `npm ci` for cleaner and more reliable node module installation.
- Enhancements to README.md to reflect the new changes in the setup instructions and provide clarity on accessing the API documentation.
- Adjustments to `ligo` compilation scripts and storage files to streamline contract generation processes.
- Removed `*_storage.gz` as it is output of test scripts.

## Additional Context
This PR addresses a bug where users were unable to create a DAO using the self-service option, as reported in issue [#767](https://app.zenhub.com/workspaces/tezos-homebase-63238426ddf385159a4c4b12/issues/gh/dorgtech/homebase-app/767). The failure was accompanied by a generic error message and a syntax error displayed in the console.

### CODE REVIEW

**Comment by ashutoshpw** [comment]
@Man-Jain, files in ligo/src/*.mligo and ligo/out/* were changed while I was running make all. I’m not sure if I should revert them to their older versions.

What do you think about adding ligo/out/ in `.gitignore` ?

File: ligo/src/implementation.mligo
Comment (Man-Jain) [inline ligo/src/implementation.mligo:1]: Where are these changes coming from? Is it a newer version of the contracts ? If not we should not change anything in here

**Review by Man-Jain — CHANGES_REQUESTED** [review]
Posted some commecnt, other then that looks good'

**Comment by ashutoshpw** [comment]
Closed in favor of #8

**Pull request closed without merging**